Add ports definition for webserver

sylvesterdamgaard opened the pull request:
**What type of PR is this?**

/kind bug

/area falco-chart

**What this PR does / why we need it**:

This PR adds a ports: section to the Falco container when the webserver/metrics settings are enabled. Without this change, the port isn’t properly exposed, preventing tools like Prometheus or Grafana from scraping metrics. By exposing the webserver port in the Pod spec, other services can now access it as intended.

**Which issue(s) this PR fixes**:

Fixes #812

**Special notes for your reviewer**:

No special notes. I’ve already updated the corresponding labels on GitHub.

**Checklist**
- [x] Chart Version bumped
- [x] ~~Variables are documented in the README.md~~
- [x] CHANGELOG.md updated

poiana commented:
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/falcosecurity/charts/pull/813#" title="Author self-approved">sylvesterdamgaard</a>*
**Once this PR has been reviewed and has the lgtm label**, please assign [issif](https://github.com/issif) for approval. For more information see [the Kubernetes Code Review Process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process).

The full list of commands accepted by this bot can be found [here](https://prow.falco.org/command-help?repo=falcosecurity%2Fcharts).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/falcosecurity/charts/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":["issif"]} -->

poiana commented:
Welcome @sylvesterdamgaard! It looks like this is your first PR to falcosecurity/charts 🎉

Issif commented:
Hi,

This is a good point.

FYI, a service is created and this the recommended endpoint to scrape.

The CI is failing for 2 reasons, you're not responsible of the failure of the `Links / linkChecker`, we did changes of the docs and some links have been migrated, we'll take care of that, you'll might have to rebase. I'll tell you when it's fixed.

For the step `Check Helm Docs / readme`, can you run `make docs-falco` to update the README to reflect your changes, thanks.

leogr commented:
> Hi,
> 
> This is a good point.
> 
> FYI, a service is created and this the recommended endpoint to scrape.
> 
> The CI is failing for 2 reasons, you're not responsible of the failure of the `Links / linkChecker`, we did changes of the docs and some links have been migrated, we'll take care of that, you'll might have to rebase. I'll tell you when it's fixed.
> 
> For the step `Check Helm Docs / readme`, can you run `make docs-falco` to update the README to reflect your changes, thanks.

For some links, we need to add a redirect in the upstream website; for others, we just have to update them here.
I'm on it. I will keep you posted.

leogr commented:
Fixed by https://github.com/falcosecurity/charts/pull/814

@sylvesterdamgaard, May I ask you to rebase this PR, please?

In particular, pay attention to use the correct rebasing procedure (to avoid merge commits): https://github.com/falcosecurity/.github/blob/main/CONTRIBUTING.md#resolving-conflicts-by-rebasing

If you are unsure or need any help, let me know.

Thanks in advance :pray:

poiana commented:
Adding label `do-not-merge/contains-merge-commits` because PR contains merge commits, which are not allowed in this repository.
Use `git rebase` to reapply your commits on top of the target branch. Detailed instructions for doing so can be found [here](https://git.k8s.io/community/contributors/guide/github-workflow.md#4-keep-your-branch-in-sync).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository.
</details>

sylvesterdamgaard commented:
@leogr I believe the PR should be up to code now. 
Rebased to remove the commit clutter and updated Changelog for a v4.17.2 release.

@Issif There was no changes to the readme since nothing changed in terms of settings.

Issif commented:
you have to update the version in the Chart.yaml 

![image](https://github.com/user-attachments/assets/4b1b2893-b74c-40c4-b506-2e19c52603a6)

sylvesterdamgaard commented:
> you have to update the version in the Chart.yaml
> 
> ![image](https://private-user-images.githubusercontent.com/4520100/404819173-4b1b2893-b74c-40c4-b506-2e19c52603a6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzczNjY1NzgsIm5iZiI6MTczNzM2NjI3OCwicGF0aCI6Ii80NTIwMTAwLzQwNDgxOTE3My00YjFiMjg5My1iNzRjLTQwYzQtYjUwNi0yZTE5YzUyNjAzYTYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDEyMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAxMjBUMDk0NDM4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZDlhZWJiYTE3ODNmMGQ0MGFhNTc0YWZlYTM0ZWFlMWIxYTZmOGY1MjI0NzAzYmM5ZDk3MTkyYmI4ZWM3MTRjOSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.rSkiA_MZNR4dg7YHcBLoinlJrticsx4Nu1HWlTIVm84)

@Issif Good catch. Updated and pushed.

Issif commented:
Now you have update the Chart.yaml, it requires the `make docs-falco` to keep updated the README, see the template: https://github.com/falcosecurity/charts/blob/afe3b77b7f5396cde437494a674a0f1e785c0a1d/charts/falco/README.gotmpl#L592

sylvesterdamgaard commented:
@Issif Sorry about that. I will get the flow eventually! :)

poiana commented:
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/falcosecurity/charts/pull/813#pullrequestreview-2562367920" title="Approved">Issif</a>*, *<a href="https://github.com/falcosecurity/charts/pull/813#" title="Author self-approved">sylvesterdamgaard</a>*

The full list of commands accepted by this bot can be found [here](https://prow.falco.org/command-help?repo=falcosecurity%2Fcharts).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/falcosecurity/charts/blob/master/OWNERS)~~ [Issif]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":[]} -->

poiana commented:
LGTM label has been added.  <details>Git tree hash: c06ab926e2b96ee75b9855c27ec781620b2c170c</details>
